### PR TITLE
forces sidebarPlugins to unmount and mount during an ActivityTab update

### LINKED ISSUE #1731: Sidebar Comments Field saves the comment on the origin File not the selected one
The `ActivityCommentAction` is not updated if `ActivityTab` get updated with new FileInfo.

## Steps to reproduce

1. Go To File View
2. Open File Details
3. Change File Details - without closing the old one
4. Create Comment

## Problem

There is a problem with the `:key` attribute.
https://github.com/nextcloud/activity/blob/22bf3f2329e5c1e7cf2b9fa627a59c02ae8ae86a/src/views/ActivityTab.vue#L17-L21

Changing it to use also something unique like `fileInfo?.id` helped:
```diff
				<ActivitySidebarPlugin v-for="plugin,index of sidebarPlugins"
-					:key="index"
+					:key="index + fileInfo?.id"
					:plugin="plugin"
					:file-info="fileInfo"
					@reload-activities="getActivities()" />
```

But neither has `ActivitySidebarPlugin` nor has `IActivitySidebarAction` an update method. :no_entry: 

So in my opinion the best option is to reproduce same behaviors as the close Button has.

1. Unmount activitySidebarPlugins
2. Let the HTML fragments get destroyed
3. Mount fresh activitySidebarPlugins

### PR DESCRIPTION
**Pull request by Messj1**
sidebarPlugins is initialized as empty array and during next tick reattached so vue force a remount.

Fixes  #1731

### CODE REVIEW

File: src/views/ActivityTab.vue [hunk 1]
```diff
@@ -91,7 +91,15 @@ export default {
...
-			this.fileInfo = fileInfo
+      this.sidebarPlugins = []
+      let sidebarPlugins = getSidebarActions()
+      if(sidebarPlugins.length > 0){
+        this.$nextTick(() => {
+          this.sidebarPlugins = sidebarPlugins
+        })
+      }
+
+      this.fileInfo = fileInfo
```
Comment (susnux) [inline src/views/ActivityTab.vue:102]: Please indent using tabs.

**Comment by Messj1** [comment]
@susnux How can I resolve this conflict with the generated file :/

After a merge of master into feature it should not appear as conflict anymore ...

```bash
$ git merge-base master bugfix/Keep-ActivitySidebarPlugin-updated 
46653414006d853a93c45464c12d616028bd6e68
$ git log -1 46653414006d853a93c45464c12d616028bd6e68
commit 46653414006d853a93c45464c12d616028bd6e68 (upstream/master, upstream/HEAD, origin/master, master)
Author: Nextcloud bot <bot@nextcloud.com>
Date:   Mon Jul 22 00:23:26 2024 +0000

    Fix(l10n): Update translations from Transifex
    
    Signed-off-by: Nextcloud bot <bot@nextcloud.com>

```

So since the `master` branch has no changes to himself, there is IMHO no reason for any conflict.

Any Ideas?

File: src/views/ActivityTab.vue [hunk 2]
```diff
@@ -91,6 +91,14 @@ export default {
 		 * @param {object} fileInfo the current file FileInfo
 		 */
 		async update(fileInfo) {
+			this.sidebarPlugins = []
+			let sidebarPlugins = getSidebarActions()
+			if(sidebarPlugins.length > 0){
```
[the remark this answers is not in the document]
Comment (Messj1) [reply src/views/ActivityTab.vue:99]: Strange things happens. At last `git rebase` was hanging in my IDE ... :open_mouth: 
With direct CLI calls it worked like expected.

BTW: The merge was only for testing purpose. :unamused:

**Review by susnux — APPROVED** [review]

**Comment by artonge** [comment]
CI failures are unrelated, merging.

**Pull request merged**